gensim/utils.py

```diff
@@ -135,7 +134,6 @@ def __exit__(self, type, value, traceback):
 nocm = NoCM()
 
 
-@contextmanager
 def file_or_filename(input):
```

Comment:
As I see in usage, this function used as contextmanager, why you remove this (`@contextmanager` + `yield`) ?

Reply:
Oh, it really doesn't work as was expected, nice catch!

Reply:
After meged comment:

Actually, it was already brokenl.  Compare with code:
```
from contextlib import contextmanager

@contextmanager
def g():
    print("g input")
    yield
    print("g exit")

@contextmanager
def f():
    print("f input")
    yield g()
    print("f exit")


with f():
    print("with block")
```

This code prints:
```
f input
with block
f exit
```

But expected something like:
f input
g input
with block
g exit
f exit
```

Back to the `file_or_filename`, methods `__enter__` and `__exit__` of `smart_open(input)` or `input` are not called at all. Oops.